common/types/shared.go

```diff
@@ -3060,6 +3684,185 @@ func (v *HistoryEvent) GetUpsertWorkflowSearchAttributesEventAttributes() (o *Up
 	return
 }
 
+// Size is an internal method to get the estimated size of the event
+func (v *HistoryEvent) Size() uint64 {
```

Comment:
We should somehow make this future proof so when someone introduces a new attribute the corresponding unit test fails because of size mismatch.